File: 0049-group-anagrams/0049-group-anagrams.cpp
class Solution {
public:
    vector<vector<string>> groupAnagrams(vector<string>& strs) {
        unordered_map<string, vector<string>> mp;

        //Loop through each word
        for (string s : strs) {
            string sorted_s = s;
            sort(sorted_s.begin(), sorted_s.end()); //sort characters

            mp[sorted_s].push_back(s); //group by sorted key
        }

        //Extract all grouped anagrams
        vector<vector<string>> result;
        for (auto& entry : mp) {
            result.push_back(entry.second);
        }

        return result;
    }
};

File: 0049-group-anagrams/0049-group-anagrams.py
class Solution(object):
    def groupAnagrams(self, strs):
        mp = {} #key : sorted string, value: list of anagrams

        # Loop through each word
        for s in strs:
            sorted_s = ''.join(sorted(s)) # sort characters to form key

            if sorted_s not in mp:
                mp[sorted_s] = []
            
            mp[sorted_s].append(s) #add original word to its group
        
        return list(mp.values())